pkg/operator/controller/controllerregistrar/reconciler.go

```diff
@@ -24,10 +25,13 @@ type Reconciler struct {
 
 // Controller contains a function for registering a controller.
 type Controller struct {
-	AddToManagerFunc func(context.Context, manager.Manager, *operatorv1alpha1.Garden) error
+	AddToManagerFunc func(context.Context, manager.Manager, *operatorv1alpha1.Garden) (bool, error)
```

Comment:
It didn't feel like a natural choice for me to return `(reconcile.Result, error)` when not being directly in a `Reconciler`.
I was more looking for a way to gracefully re-enqueue in case a `AddToManagerFunc` needs to wait longer (similar to or inspired by `retry.Until`), with a common `RequeueAfter`.